hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -129,6 +129,9 @@ public static ClientMessage create() {
 
     public static ClientMessage createForEncode(int initialCapacity) {
         initialCapacity = QuickMath.nextPowerOfTwo(initialCapacity);
+        if (initialCapacity < 0) {
+            initialCapacity = Integer.MAX_VALUE;
+        }
         if (USE_UNSAFE) {
```

Comment:
I do not think so. the point of the `QuickMath` class is to do as little work as possible. Its contract explicitly says:
```
This method is not suitable for {@link Integer#MIN_VALUE} or numbers greater than 2^30.
```